bots/heroes/plague_main.lua

```diff
@@ -0,0 +1,355 @@
+	= _G.print, _G.ipairs, _G.pairs, _G.string, _G.table, _G.next, _G.type, _G.table.insert, _G.table.remove, _G.table.sort, _G.string.format, _G.tostring, _G.tonumber, _G.string.find, _G.string.sub
+local ceil, floor, pi, tan, atan, atan2, abs, cos, sin, acos, max, random
+	= _G.math.ceil, _G.math.floor, _G.math.pi, _G.math.tan, _G.math.atan, _G.math.atan2, _G.math.abs, _G.math.cos, _G.math.sin, _G.math.acos, _G.math.max, _G.math.random
+
+local BotEcho, VerboseLog, BotLog = core.BotEcho, core.VerboseLog, core.BotLog
+local Clamp = core.Clamp
+
+BotEcho('loading plague_main...')
+
+---------------------------------------------------
+--                  Constants
+---------------------------------------------------
+
+object.heroName = 'Hero_DiseasedRider'
+
+-- Lanes
+core.tLanePreferences = {Jungle = 0, ShortCarry = 1, LongCarry = 1, Mid = 2, ShortSolo = 2, LongSolo = 3, ShortSupport = 5, LongSupport = 5}
+
+-- Skillbuild table, 0 = q, 1 = w, 2 = e, 3 = r, 4 = attri
+object.tSkills = {
+	0, 2, 0, 2, 0,
+	3, 0, 2, 2, 1,
+	3, 1, 1, 1, 4,
+	3, 4, 4, 4, 4,
+	4, 4, 4, 4, 4
+}
+
+----------------------------------
+--	Plague items
+----------------------------------
+behaviorLib.StartingItems = {"Item_PretendersCrown", "2 Item_MinorTotem", "Item_HealthPotion", "2 Item_RunesOfTheBlight"}
+behaviorLib.LaneItems = 
+	{"Item_Strength5", "Item_Striders", "Item_MysticPotpourri", "Item_MysticVestments"} -- Strength5 is Fortified Bracer and Potpourri is Refreshing Ornament
+behaviorLib.MidItems = 
+	{"Item_Astrolabe", "Item_Intelligence7", "Item_PostHaste", "Item_Protect"} -- Intelligence7 is SotM and Protect is Null Stone
+behaviorLib.LateItems = 
+	{"Item_Morph", "Item_BehemothsHeart", "Item_Immunity"} -- Morph is Sheepstick
+
+--------------------------------
+-- Skills
+--------------------------------
+local bSkillsValid = false
+function object:SkillBuild()
+
+	local unitSelf = self.core.unitSelf
+    if not bSkillsValid then
+		skills.abilNuke = unitSelf:GetAbility(0)
+        skills.abilShield = unitSelf:GetAbility(1)
+        skills.abilMana = unitSelf:GetAbility(2)
+        skills.abilUltimate = unitSelf:GetAbility(3)
+		
+		if (skills.abilNuke and skills.abilShield and skills.abilMana and skills.abilUltimate) then
+			bSkillsValid = true
+		else
+			return
+		end		
+    end
+    if unitSelf:GetAbilityPointsAvailable() <= 0 then
+        return
+    end    
+   
+    local nlev = unitSelf:GetLevel()
+    local nlevpts = unitSelf:GetAbilityPointsAvailable()
+    for i = nlev, nlev+nlevpts do
+        unitSelf:GetAbility( object.tSkills[i] ):LevelUp()
+    end
+end
+
+---------------------------------------------------
+--                   Utilities                   --
+---------------------------------------------------
+
+-- bonus aggression points if a skill/item is available for use
+object.nNukeUp = 18
+object.nUltimateUp = 7
+-- bonus aggression points that are applied to the bot upon successfully using a skill/item
+object.nNukeUse = 14

```

Comment:
Using nuke actually drops aggression by 4 points as nNukeUp is no longer added cause it is on cooldown after the cast.